autosklearn/ensemble_builder.py

```diff
@@ -692,25 +696,71 @@ def predict(self, set_: str,
 
     def _delete_non_candidate_models(self, candidates):
         candidates = [os.path.split(cand)[1] for cand in candidates]
-        for model_path in self.read_preds.keys():
-            if self.read_preds[model_path]['deleted']:
+        for pred_path in self.read_preds.keys():
+            if self.read_preds[pred_path]['deleted']:
                 continue
-            match = self.model_fn_re.search(model_path)
-            _num_run = int(match.group(2))
+            match = self.model_fn_re.search(pred_path)
+            _seed = match.group(1)
+            _num_run = match.group(2)
+            _budget = match.group(3)
             # Do not remove the dummy prediction!
-            if _num_run == 1:
+            if int(_num_run) == 1:
                 continue
-            model_file = os.path.split(model_path)[1]
-            if model_file not in candidates:
+            pred_file = os.path.split(pred_path)[1]
+            if pred_file not in candidates:
+                # Messages logged bellow should match what is asserted
+                # in test_delete_non_candidate_models(). Please, ensure consistency.
+
+                # Delete prediction file
+                try:
+                    os.remove(pred_path)
+                    self.logger.info(
+                        'Deleted prediction file of non-candidate model %s', pred_path)
+                    self.read_preds[pred_path]['deleted'] = True
```

Comment:
I'm wondering whether we should add a flag for each of the files that can/must be deleted. That would increase chances of deleting all files in case something weird happens (a file being locked). What do you think?

Reply:
I think it is a good idea. We should talk more about it tomorrow.